specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,422 @@
+
+  /// Get the `SupressDefaultDialog`
+  [propget] HRESULT SupressDefaultDialog([out, retval] BOOL* value);
+
+  /// Returns an `ICoreWebView2Deferral` object. This will defer showing the 
+  /// default Save As dialog and performing the Save As operation.
+  HRESULT GetDeferral([out, retval] ICoreWebView2Deferral** deferral);
+
+  /// `ResultFilePath` is absolute full path of the location. It includes the file name
+  /// and extension. If `ResultFilePath` is not valid, for example the root drive does
+  /// not exist, save as will be denied and return COREWEBVIEW2_SAVE_AS_INVALID_PATH.
+  ///
+  /// If the associated download completes successfully, a target file will be saved at 
+  /// this location. If the Kind property is `COREWEBVIEW2_SAVE_AS_REQUESTED_KIND_COMPLETE`, 
+  /// there will be an additional directory with resources files.
+  ///
+  /// The default value is a system suggested path, based on users' local environment.
+  /// 
+  /// Set the `ResultFilePath` for save as
+  [propput] HRESULT ResultFilePath ([in] LPCWSTR value);
+
+  /// Get the `ResultFilePath` for save as
+  [propget] HRESULT ResultFilePath ([out, retval] LPWSTR* value);
+
+  /// `AllowReplace` allows you to control what happens when a file already 
+  /// exists in the file path to which the Save As operation is saving.
+  /// Setting this TRUE allows existing files to be replaced.
+  /// Settings this FALSE will not replace existing files and  will return
+  /// COREWEBVIEW2_SAVE_AS_FILE_ALREADY_EXISTS.
+  ///
+  /// The default value is FALSE
+  ///
+  /// Set if allowed to replace the old file if duplicate happens in the save as
+  [propput] HRESULT AllowReplace ([in] BOOL value);
+
+  /// Get the duplicates replace rule for save as
+  [propget] HRESULT AllowReplace ([out, retval] BOOL* value);
+
+  /// How to save documents with different kind. See the enum 
+  /// COREWEBVIEW2_SAVE_AS_REQUESTED_KIND for a description of the different options.  
+  /// If the kind isn't allowed for the current document, 
+  /// COREWEBVIEW2_SAVE_AS_REQUESTED_KIND_NOT_SUPPORT will be returned from RequestSaveAs.
+  ///
+  /// The default value is COREWEBVIEW2_SAVE_AS_REQUESTED_KIND_DEFAULT
+  ///
+  /// Set the kind for save as
+  [propput] HRESULT Kind ([in] COREWEBVIEW2_SAVE_AS_REQUESTED_KIND value);
+
+  /// Get the kind for save as
+  [propget] HRESULT Kind ([out, retval] COREWEBVIEW2_SAVE_AS_REQUESTED_KIND* value);
+}
+
+/// Receive the result for `RequestSaveAs` method
+[uuid(1a02e9d9-14d3-41c6-9581-8d6e1e6f50fe), object, pointer_default(unique)]
+interface ICoreWebView2RequestSaveAsCompletedHandler : IUnknown {
+  HRESULT Invoke([in] HRESULT errorCode, [in] COREWEBVIEW2_REQUEST_SAVE_RESULTS result);
+}
+```
+
+## .Net/ WinRT
+```c# (but really MIDL3)
+namespace Microsoft.Web.WebView2.Core
+{
+
+    runtimeclass CoreWebView2SaveAsRequestedEventArgs; 
+    runtimeclass CoreWebView2;
+
+    enum CoreWebView2SaveAsRequestedResults
```

Comment:
Singular, since this is not a [flags] enum
```suggestion
    enum CoreWebView2SaveAsRequestedResult
```

Reply:
Please rename to names chosen above.